pkg/certificate/providers/vault/certificate_manager.go

```diff
@@ -30,23 +25,13 @@ const (
 	commonNameField   = "common_name"
 	ttlField          = "ttl"
 
-	checkCertificateExpirationInterval = 5 * time.Second
-	decade                             = 8765 * time.Hour
+	decade = 8765 * time.Hour
 )
 
-// NewCertManager implements certificate.Manager and wraps a Hashi Vault with methods to allow easy certificate issuance.
-func NewCertManager(
-	vaultAddr,
-	token string,
-	role string,
-	cfg configurator.Configurator,
-	serviceCertValidityDuration time.Duration,
-	msgBroker *messaging.Broker) (*CertManager, error) {
+// New will construct a new certificate client using Vault's cert-manager
+func New(vaultAddr, token string, role string) (*CertManager, error) {
```

Comment:
nit: if you're specifying the param type once for 2 variables, may as well do once for all 3

Reply:
I actually love this nit, good call out.